bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/CoreItemFactory.java

```diff
@@ -27,69 +27,62 @@
  *
  * @author Thomas.Eichstaedt-Engelen
  * @author Kai Kreuzer
+ * @author Alexander Kostadinov
  */
 public class CoreItemFactory implements ItemFactory {
 
-    public static final String SWITCH = "Switch";
-    public static final String ROLLERSHUTTER = "Rollershutter";
+    public static final String CALL = "Call";
+    public static final String COLOR = "Color";
     public static final String CONTACT = "Contact";
-    public static final String STRING = "String";
-    public static final String NUMBER = "Number";
-    public static final String DIMMER = "Dimmer";
     public static final String DATETIME = "DateTime";
-    public static final String COLOR = "Color";
+    public static final String DIMMER = "Dimmer";
     public static final String IMAGE = "Image";
-    public static final String PLAYER = "Player";
     public static final String LOCATION = "Location";
-    public static final String CALL = "Call";
+    public static final String NUMBER = "Number";
+    public static final String PLAYER = "Player";
+    public static final String ROLLERSHUTTER = "Rollershutter";
+    public static final String STRING = "String";
+    public static final String SWITCH = "Switch";
 
     /**
-     * @{inheritDoc
+     * {@inheritDoc}
      */
     @Override
     public GenericItem createItem(String itemTypeName, String itemName) {
-        if (itemTypeName.equals(SWITCH)) {
-            return new SwitchItem(itemName);
-        }
-        if (itemTypeName.equals(ROLLERSHUTTER)) {
-            return new RollershutterItem(itemName);
-        }
-        if (itemTypeName.equals(CONTACT)) {
-            return new ContactItem(itemName);
-        }
-        if (itemTypeName.equals(STRING)) {
-            return new StringItem(itemName);
-        }
-        if (itemTypeName.equals(NUMBER)) {
-            return new NumberItem(itemName);
-        }
-        if (itemTypeName.equals(DIMMER)) {
-            return new DimmerItem(itemName);
-        }
-        if (itemTypeName.equals(DATETIME)) {
-            return new DateTimeItem(itemName);
-        }
-        if (itemTypeName.equals(COLOR)) {
-            return new ColorItem(itemName);
-        }
-        if (itemTypeName.equals(IMAGE)) {
-            return new ImageItem(itemName);
-        }
-        if (itemTypeName.equals(PLAYER)) {
-            return new PlayerItem(itemName);
-        }
-        if (itemTypeName.equals(LOCATION)) {
-            return new LocationItem(itemName);
-        }
-        if (itemTypeName.equals(CALL)) {
-            return new CallItem(itemName);
-        } else {
-            return null;
+        if (itemTypeName != null) {
```

Comment:
Good you thought about it! 
Please just invert this statement in order to reduce the nesting level.